### PR TITLE
l10n cleanup

### LINKED ISSUE #204: Improvements to fluent usage
Improve how Fluent is used in the addon to better help localizers out.

Summarized from a communique via @flodolo:

* brand names should be terms, not standard messages. You can add a section comment for them ([example](https://hg.mozilla.org/l10n/gecko-strings/file/default/browser/branding/official/brand.ftl))
* No need to add NOTE: to comments
* It would be nice to avoid unnecessarily new lines when the string is short
* We shouldn't have select expressions special casing 0 for something different from the other cases. We should have a string for the zero case, and handle the rest in the code. ([incorrect example](https://github.com/mozilla-lockbox/lockbox-addon/blob/master/src/locales/en-US/list.ftl#L44-L48))
* Section comment should be explanatory and help localizers.
* Resolve all the TODOs
* Add l10n reviewer to CODEOWNERS

## To Do

* [ ] Separate unlocalizable strings and localizable strings into separate files
* [x] Ensure the following are (unlocalizable) terms:
  - Firefox
  - (Firefox) Lockbox|Lockwise
  - (Firefox) Sync
* [x] Note that "Firefox Account" requires [special treatment](https://hg.mozilla.org/l10n/gecko-strings/file/default/browser/browser/branding/sync-brand.ftl#l11)
  - [x] Should use `-fxaccount-brand-name` message ID to align with other projects
* [x] Consistent use of the following **localizable** terms:
  - `add-on`
* [ ] Use self-explanatory placeables/slugs; e.g., `-fxlockwise-product-name` instead of `-product-name`
* [x] Consistent spacing of slugs and placeables; e.g. `{ $count }` and `{ -product-full-name }` instead of `{$count}` or `{product-full-name}`
* [x] Fix/add section names to be more descriptive of what/where the strings are used for
* [x] Remove references/pointers to (now nonexistent) [translation guidelines](https://www.mozilla.org/styleguide/communications/translation/)
* [x] Remove any "column alignment" spacers; e.g., do the following:
```
list-count = ( $count ->
  [one] { $count } entry
 *[other] { $count } entries
}
```
* [ ] Comment on (or otherwise remove) seemingly idential strings to help explain differences:
  - `connect-a-firefox-account` vs `connect-a-firefox-account-complete`
* [ ] Make sure the app/play store icons have separate LICENSE files
* [ ] Change the promo banner CTA string (`banner-promote-device .details` and `banner-promote-fxa .details`) to be one string with an infixed `<strong>` tag.
  - see bedrock examples listed [here](https://github.com/mozilla-lockwise/lockwise-addon/pull/209#issuecomment-490754969)

### PR DESCRIPTION
**Pull request by 6a68**
Fixes #204 (addresses remaining unchecked checkboxes, except for splitting untranslated and translated content, which I think we've handled via comments in list.ftl).

Feedback welcome on the handling of the string in the modal. Infixing the strong tag led to JSX trimming the whitespace on the right side of the tag, so I resorted to including a margin in the strong tag. I can move that into a classname, if desired.

Not sure what's up with the broken unit test; I'll look at that in a bit.

## To Do

- add “WIP” to the PR title if pushing up but not complete nor ready for review
- [ ] double check the original issue to confirm it is fully satisfied
- [ ] add testing notes and screenshots in PR description to help guide reviewers
- [ ] add unit tests
  - optional: consider adding integration tests
- [ ] request the "UX" team perform a design review (if/when applicable)
- [ ] make sure CI builds are passing (e.g.: fix lint and other errors)
- [ ] check on the [accessibility](https://mozilla-lockbox.github.io/lockbox-addon/developer/test-plan-accessibility/) of any added UI
- [ ] make sure any new UI has telemetry events wired up and documented in docs/metrics.md, and verify that the events are recording properly (visit `about:telemetry#events-tab`, then choose 'dynamic' from the dropdown menu at top right, to view events fired by the addon

### CODE REVIEW

**Comment by 6a68** [comment]
rebased.

**Comment by 6a68** [comment]
Ah, the buttons aren't rendering. Going to have to tweak some more.

**Comment by 6a68** [comment]
@linuxwolf thanks for the feedback, think I've got it sorted. wound up keeping the LocalizedBanner just to eliminate duplication between the two views

File: src/locales/en-US/list.ftl [hunk 1]
```diff
@@ -195,26 +195,22 @@ easily-access-logins = Easily gain access to your logins from any device.
...
 before-access = Before you can access your logins on another device, you will need to connect a { -fxaccount-brand-name }.
 connect-a-firefox-account = Connect a { -fxaccount-brand-name }
-# Displayed after the connection has completed
-connect-a-firefox-account-complete = Connect a { -fxaccount-brand-name } (complete)
-sync-requires-account = To sync your logins to another device, you will need to <signin>sign in or create a { -product-full-name }</signin>.
+# Displayed next to connect-a-firefox-account string after the user has logged in.
+connection-complete = (complete)
+sync-requires-account = To sync your logins to another device, you will need to <signin>sign in or create a { -fxlockwise-brand-name }</signin>.
 ensure-logins-checked = Ensure the “Logins” checkbox is selected in { -sync-brand-short-name } preferences
 setting-to-allow-sync = In order to allow your logins to be synced to other devices, this setting must be checked. <openprefs>Open { -sync-brand-short-name } preferences</openprefs></p>
```
Comment (pdehaan) [inline src/locales/en-US/list.ftl]: Unrelated to this PR, but it looks like this string has a closing `</p>` but no opening `<p>`.

**Pull request merged**